src/sentry/taskworker/registry.py

```diff
@@ -126,19 +126,38 @@ def wrapped(func: Callable[P, R]) -> Task[P, R]:
         return wrapped
 
     def send_task(self, activation: TaskActivation, wait_for_delivery: bool = False) -> None:
-        metrics.incr("taskworker.registry.send_task", tags={"namespace": activation.namespace})
-
         topic = self.router.route_namespace(self.name)
         produce_future = self._producer(topic).produce(
             ArroyoTopic(name=topic.value),
             KafkaPayload(key=None, value=activation.SerializeToString(), headers=[]),
         )
+        try:
+            if produce_future.exception(timeout=1):
```

Comment:
Instead of blocking on an produce returning an exception/value, should we be using `add_done_callback()` and generating the metrics asynchronously?

Reply:
Ah yeah, that's a better way of doing this. I'm still using `future.exception(1)` to capture the metric failure metric, but it doesn't block anymore since it's part of the callback when the future finishes and exception gets set.